src/poll.rs

```diff
@@ -537,22 +543,23 @@ struct AtomicState {
 
 const MASK_2: usize = 4 - 1;
 const MASK_4: usize = 16 - 1;
+const MASK_8: usize = 256 - 1;
 const QUEUED_MASK: usize = 1 << QUEUED_SHIFT;
 const DROPPED_MASK: usize = 1 << DROPPED_SHIFT;
 
 const READINESS_SHIFT: usize = 0;
-const INTEREST_SHIFT: usize = 4;
-const POLL_OPT_SHIFT: usize = 8;
-const TOKEN_RD_SHIFT: usize = 12;
-const TOKEN_WR_SHIFT: usize = 14;
-const QUEUED_SHIFT: usize = 16;
-const DROPPED_SHIFT: usize = 17;
+const INTEREST_SHIFT: usize = 8;
+const POLL_OPT_SHIFT: usize = 16;
+const TOKEN_RD_SHIFT: usize = 20;
+const TOKEN_WR_SHIFT: usize = 22;
+const QUEUED_SHIFT: usize = 24;
+const DROPPED_SHIFT: usize = 25;
```

Comment:
Ok.  BTW, I know of three possible ways to do AIO with mio on Linux, but none of them is very good:
1) Use POSIX AIO.  Use a separate thread to call `aio_suspend` (similar to `select` but strictly for POSIX AIO).  When it returns, shove an event into mio's custom readiness queue.  This is the slowest option.
2) Use LibAIO.  Use a separate thread to call `io_getevents`.  When it returns, shove an event into mio's custom readiness queue.
3) Use LibAIO with a signalfd.  Register the signalfd with mio.  When it's ready, read a `signalfd_siginfo` out of it.  Either shove an AIO event into mio's custom readiness queue, or perhaps translate it inline.  This is probably the best approach, though there are complications relating to signal delivery.